crates/subspace-farmer/src/archiving.rs

```diff
@@ -125,13 +125,6 @@ impl Archiving {
             while let Some(archived_segment) = archived_segments.next().await {
                 let segment_index = archived_segment.root_block.segment_index();
                 let (acknowledge_sender, acknowledge_receiver) = oneshot::channel();
-                // Acknowledge immediately to allow node to continue sync quickly,
-                // but this will miss some segments in case farmer crashed in the
-                // meantime. Ideally we'd acknowledge after, but it makes node wait
-                // for it and the whole process very sequential.
-                if let Err(error) = client.acknowledge_archived_segment(segment_index).await {
```

Comment:
Previously we were waiting for farmer to acknowledge and plot the segment before proceeding, otherwise due to max plot size restriction farmer may store too many pieces that are actually outside of current max plot size.

This restriction doesn't apply in V2 consensus anymore as we don't have max plot size notion and we determine which pieces should be plotted based on identity, sector and current size of the blockchain history.

However, there might still be a need to pause block import in certain cases to make sure things happen in lock step, for that acknowledgement can still be useful, though we'll likely want to either drop this feature entirely or introduce a separate subscription for such use case specifically. I decided to keep it for now, but not block by farmer anymore.